### DISCUSSION TITLE
Support all java.time data types

### DISCUSSION THREAD

**Discussion by mhinterreither** [mapstruct/mapstruct#3076] [Ideas]
Hello,
I see that mapstruct has integrated support for the mapping of some data types from java.time package from/to String e.g. java.time.ZonedDateTime. But some other data types from that package are not supported out of the box to be mapped from/to String e.g. java.time.OffsetDateTime.
Is there a reason why java.time.OffsetDateTime (and some others) from/to String mapping isn't implemented directly in mapstruct or would it make sense to add this in future versions of mapstruct?

**Comment by claui** [comment]
+1 from me. That would absolutely make sense.

Networked applications usually need to conform to [RFC 3339](https://www.rfc-editor.org/rfc/rfc3339), which supports time zone offsets but not Java’s notion of full time-zone rules. That same RFC is also enforced by JSON Schema. So any Java application that uses JSON would typically use `OffsetDateTime` for the backing model, not `ZonedDateTime`.

**Comment by filiphr** [comment]
The only reason why we haven't implemented something like this is because no one as asked for it and we haven't seen the need for it. However, adding functionality for more implicit conversions is easy to add and we are more than happy to accept contributions for this.

@mhinterreither or @claui will you be interested in adding conversions that you think MapStruct should have for `java.time` data types?
